### ISSUE TITLE
Mercurial certificates not properly configured on macOS

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#3847]
I'm trying to build mercurial on macOS but it seems like the certificates aren't properly configured:
```
hg clone https://bitbucket.org/MatthewTurk/rockstar
(an attempt was made to load CA certificates but none were loaded; see https://mercurial-scm.org/wiki/SecureConnections for how to configure Mercurial to avoid this error)
abort: error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:661)
```
This problem was solved for Linux by adding the certificate to `hgrc`, but it doesn't seem to work for macOS. Note that:
https://www.mercurial-scm.org/wiki/CACertificates#Mac_OS_X_10.6_and_higher
has several suggestions for how to solve this problem, but none of them seem to work. They claim that:

> Mercurial 3.2 or later enables the dummy certificate trick by default.

but it doesn't seem to work for our Spack build. When I build mercurial outside of Spack with ananconda python, it works. But when I link to Spack's Python build, it no longer works. Maybe Python isn't built properly either?

**Comment by adamjstewart** [comment]
The relevant section of source code is: https://www.mercurial-scm.org/repo/hg/file/tip/mercurial/sslutil.py#l687

I wonder if it's worth trying to hack this to use the dummy cert or if we should just download a cert and include it as a dependency. Thoughts?

**Comment by adamjstewart** [comment]
Homebrew's Mercurial build doesn't do anything special and it seems to work just fine. I'm so confused.
https://github.com/Homebrew/homebrew-core/blob/master/Formula/mercurial.rb

**Comment by adamjstewart** [comment]
I'm going to try using the certifi package and see if that helps:
https://www.mercurial-scm.org/repo/hg/file/tip/mercurial/sslutil.py#l697

**Comment by adamjstewart** [comment]
Well, certifi is a partial fix. Mercurial still doesn't work unless you load the `py-certifi` module, but afterwards it works.

**Comment by adamjstewart** [comment]
Oh, wait:
```
>>> import certifi
>>> certifi.where()
'/Users/Adam/spack/opt/spack/darwin-sierra-x86_64/clang-8.1.0-apple/py-certifi-2017.1.23-dkf2tvbg6cigetzp663riw3nqulxtb5w/lib/python2.7/site-packages/certifi-2017.1.23-py2.7.egg/certifi/cacert.pem'
```
This could work...

**Comment by adamjstewart** [comment]
Success! This will be fixed in #3834.

**Comment by keitat** [comment]
I did see the fix when installing xSDK on our PC cluster.
